testcases/MachineConfig.py

```diff
@@ -300,3 +317,101 @@ def CecSetup_lmb(self):
+        self.mmu = str(self.mmulist[0]).split(':')[1].strip()
+        self.obj = OpTestInstallUtil.InstallUtil()
+        self.os_level = self.cv_HOST.host_get_OS_Level()
+
+
+    def OsHugepageSetup(self):
+
+        filename_part1 = "cat /sys/kernel/mm/hugepages/hugepages-"
+        filename_part2 = "kB/nr_hugepages"
+        if "hugepage=16G" in self.machine_config:
+            self.size_hgpg = "16G"
+            self.num_hgpg = self.cv_HMC.get_16gb_hugepage_size()
+            self.no_hgpg = int(self.num_hgpg[0])
+            if self.no_hgpg != 0:
+                self.configure_16G_hugepage()
+                con = self.cv_SYSTEM.cv_HOST.get_ssh_connection()
+                assign_hp = con.run_command("%s%s%s" % (
+                    filename_part1, "16777216", filename_part2))[0]
+            else:
+                return "16gb hugepages are not configured in CEC"
+        else:
+            exist_cfg = self.cv_HMC.get_lpar_cfg()
+            self.des_mem = int(exist_cfg.get('desired_mem'))
+            self.percentile = int(self.des_mem * 0.1)
+            if 'Radix' in self.mmu:
+                if "hugepage=16M" in self.machine_config:
+                    self.fail("16M is not supported in Radix")
+                elif "hugepage=2M" in self.machine_config:
+                    self.size_hgpg = "2M"
+                    self.no_hgpg = int(self.percentile / 2)
+                elif "hugepage=1G" in self.machine_config:
+                    self.size_hgpg = "1G"
+                    self.no_hgpg = int(self.percentile / 1024)
+
+            elif 'Hash' in self.mmu and "hugepage=16M" in self.machine_config:
+                self.size_hgpg = "16M"
+                self.no_hgpg = int(self.percentile / 16)
+            self.obj.update_kernel_cmdline(self.os_level,
+                                           "hugepagesz=%s hugepages=%s" % (
+                                           self.size_hgpg, self.no_hgpg),
+                                           "",
+                                           reboot=True,
+                                           reboot_cmd=True)
+            con = self.cv_SYSTEM.cv_HOST.get_ssh_connection()
+            if self.size_hgpg == "2M":
+                assign_hp = con.run_command("%s%s%s" % (
+                    filename_part1, "2048", filename_part2))[0]
+            elif self.size_hgpg == "1G":
+                assign_hp = con.run_command("%s%s%s" % (
+                    filename_part1, "1048576", filename_part2))[0]
+            elif self.size_hgpg == "16M":
+                assign_hp = con.run_command("%s%s%s" % (
+                    filename_part1, "16384", filename_part2))[0]
+        if str(self.no_hgpg) != assign_hp:
+            msg = "Expected %s: But found %s" % (self.no_hgpg, assign_hp)
+            return msg
+        else:
+            log.info("%s Hugepage validation successful!" % self.size_hgpg)
+
+
+    def configure_16G_hugepage(self):
```

Comment:
rename to differentiate between hmc configure method and os configure method  like configure_os_hugepage